### PR TITLE
Add Helm chart

### PR DESCRIPTION
**Pull request by ellisvalentiner**
*Issue #, if available:* #41 

*Description of changes:* Adds Helm chart

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ You can create an IAM policy using this template, and attach it to the role if y
 ## Deploy
 Requires a Kubernetes cluster with Metric Server deployed, Amazon EKS cluster is fine too.
 
-Now deploy the adapter to your Kubernetes cluster.
+Now deploy the adapter to your Kubernetes cluster using Kustomize:
```
Comment (chankh) [inline README.md:51]: Not using Kustomize here

File: README.md [hunk 2]
```diff
@@ -71,6 +71,8 @@ clusterrolebinding.rbac.authorization.k8s.io/k8s-cloudwatch-adapter:crd-metrics-
 This creates a new namespace `custom-metrics` and deploys the necessary ClusterRole, Service Account,
 Role Binding, along with the deployment of the adapter.
 
+Alternatively the adapter can be deployed using the Helm chart in the `/chart` directory.
+
```
Comment (chankh) [inline README.md:75]: can add the commands for using helm too

File: chart/templates/crd.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+metadata:
+  name: k8s-cloudwatch-adapter:crd-metrics-reader
+  labels:
+    app: k8s-cloudwatch-adapter
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: k8s-cloudwatch-adapter:crd-metrics-reader
+subjects:
+  - name: k8s-cloudwatch-adapter
```
Comment (chankh) [inline chart/templates/crd.yaml:41]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+kind: Deployment
+metadata:
+  labels:
+    app: "{{ .Release.Name }}"
+  name: "{{ .Release.Name }}"
+  namespace: "{{ .Release.Namespace }}"
+  annotations:
+    {{- if .Values.iam_role }}
+    iam.amazonaws.com/role: {{ .Values.iam_role }}
+    {{- end }}
```
Comment (chankh) [inline chart/templates/deployment.yaml:11]: IAM role should be specified in the service account, as shown here https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html

File: chart/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  template:
+    metadata:
+      labels:
+        app: "{{ .Release.Name }}"
+      name: "{{ .Release.Name }}"
+      namespace: "{{ .Release.Namespace }}"
+      annotations:
+        {{- if .Values.iam_role }}
+        iam.amazonaws.com/role: {{ .Values.iam_role }}
+        {{- end }}
```
Comment (chankh) [inline chart/templates/deployment.yaml:26]: same for this

File: chart/templates/rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-cloudwatch-adapter:system:auth-delegator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+- kind: ServiceAccount
+  name: k8s-cloudwatch-adapter
```
Comment (chankh) [inline chart/templates/rbac.yaml:11]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v2
+name: k8s-cloudwatch-adapter
+description: A Helm chart for Kubernetes
```
Comment (chankh) [inline chart/Chart.yaml:3]: give it a proper description

File: chart/templates/crd.yaml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    plural: externalmetrics
+    singular: externalmetric
+  scope: Namespaced
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: k8s-cloudwatch-adapter:crd-metrics-reader
+  labels:
+    app: k8s-cloudwatch-adapter
```
Comment (chankh) [inline chart/templates/crd.yaml:19]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/crd.yaml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  - list
+  - get
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-cloudwatch-adapter:crd-metrics-reader
+  labels:
+    app: k8s-cloudwatch-adapter
```
Comment (chankh) [inline chart/templates/crd.yaml:35]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/rbac.yaml [hunk 10]
```diff
@@ -0,0 +1,77 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-cloudwatch-adapter:system:auth-delegator
```
Comment (chankh) [inline chart/templates/rbac.yaml:4]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/rbac.yaml [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+metadata:
+  name: k8s-cloudwatch-adapter-auth-reader
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: extension-apiserver-authentication-reader
+subjects:
+- kind: ServiceAccount
+  name: k8s-cloudwatch-adapter
```
Comment (chankh) [inline chart/templates/rbac.yaml:25]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/rbac.yaml [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  name: system:auth-delegator
+subjects:
+- kind: ServiceAccount
+  name: k8s-cloudwatch-adapter
+  namespace: "{{ .Release.Namespace }}"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: k8s-cloudwatch-adapter-auth-reader
```
Comment (chankh) [inline chart/templates/rbac.yaml:17]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/rbac.yaml [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-cloudwatch-adapter-resource-reader
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: k8s-cloudwatch-adapter-resource-reader
+subjects:
+- kind: ServiceAccount
+  name: k8s-cloudwatch-adapter
```
Comment (chankh) [inline chart/templates/rbac.yaml:38]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/rbac.yaml [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+  namespace: "{{ .Release.Namespace }}"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: k8s-cloudwatch-adapter:external-metrics-reader
```
Comment (chankh) [inline chart/templates/rbac.yaml:44]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/rbac.yaml [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+rules:
+- apiGroups:
+  - external.metrics.k8s.io
+  resources: ["*"]
+  verbs: ["*"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: k8s-cloudwatch-adapter-resource-reader
```
Comment (chankh) [inline chart/templates/rbac.yaml:54]: should you replace the names with `{{ .Release.Name }}` too?

File: chart/templates/rbac.yaml [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-cloudwatch-adapter:external-metrics-reader
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: k8s-cloudwatch-adapter:external-metrics-reader
```
Comment (chankh) [inline chart/templates/rbac.yaml:73]: should you replace the names with `{{ .Release.Name }}` too?

**Comment by ellisvalentiner** [comment]
@chankh incorporated the changes you requested, let me know if there's anything else

**Pull request merged**